app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java

```diff
@@ -0,0 +1,244 @@
+        notifyItemRemoved(position);
+    }
+
+    /**
+     * Update the list of contacts to show
+     * and notify that data have been updated.
+     *
+     * @param newContacts New list of contacts to show and manage
+     */
+    public void updateItems(List<SMSContact> newContacts) {
+        contacts.clear();
+        contacts.addAll(newContacts);
+        filter = new ContactFilter(this, contacts);
+        notifyDataSetChanged();
+    }
+
+
+    //---------------------------- ContactViewHolder ----------------------------
+
+    /**
+     * Class to represent a contact viewed in a row of RecycleView.
+     *
+     * @author Giorgia Bortoletti
+     */
+    public class ContactViewHolder extends RecyclerView.ViewHolder {
+        public TextView contactName;
+        public TextView contactAddress;
+        public ConstraintLayout layout;
+
+        /**
+         * Constructor
+         *
+         * @param itemView
+         */
+        public ContactViewHolder(View itemView) {
+            super(itemView);
+            contactName = itemView.findViewById(R.id.contact_name);
+            contactAddress = itemView.findViewById(R.id.contact_address);
+            layout = itemView.findViewById(R.id.contact_item);
+        }
+
+        /**
+         * Set the item
+         * in its TextView and in its Listener to select one.
+         *
+         * @param contact To set in the item.
+         */
+        public void bind(SMSContact contact, int position) {
+            if (position == selectedPosition)
+                layout.setBackgroundColor(ContextCompat.getColor(layout.getContext(), R.color.selectedContact));
+            else
+                layout.setBackgroundColor(Color.TRANSPARENT);
+            contactName.setText(contact.getName());
+            contactAddress.setText(contact.getAddress());
+
+            itemView.setOnClickListener(new View.OnClickListener() {
+                /**
+                 * Change status of clicked item.
+                 * If it had already been selected, back to be all default
+                 * otherwise background changes color and the item becomes the contact selecte.
+                 *
+                 * @param view
+                 */
+                @Override
+                public void onClick(View view) {
+                    if (selectedPosition != getAdapterPosition()) {
+                        layout.setBackgroundColor(ContextCompat.getColor(layout.getContext(), R.color.selectedContact));
+                        notifyItemChanged(selectedPosition);
+                        selectedPosition = getAdapterPosition();
+                    } else { //user has re-clicked the same item
+                        layout.setBackgroundColor(Color.TRANSPARENT);
+                        notifyItemChanged(selectedPosition);
+                        selectedPosition = DEFAULT_SELECTED_POSITION;
+                    }
+                    MainActivity.setContactSelected(getSelectedItem());
```

Comment:
Done, thank you.
Further changes will be made in the next days because Riccardo has transformed these Activities into Fragments and therefore I will update the methods in Fragment with more precise specifications.